Remove generateTT variable. TT is always generated.

polinankoleva opened the pull request:
+ some typos fixes

gschievelbein commented:
I believe that we need a variable for every model that we load. Although we always generate generate a TT with the current models that we have implemented, if we ever add a new model, or change the requirements for the current models, we will need to undo those changes. 
I think it makes more sense that we check that requirements are fulfilled instead of assuming that the TT is always loaded.

gschievelbein commented:
I suggest that we do not remove the checks for the requirements for loading the models, as this would reduce the code robustness.
A better solution is to then check all dependencies after all input parameters are read instead. There are other checks with the stats and dropDuplicates parameters. This way all requirements checks are done in one place.
I also would not remove the parameter to enable the loading of TT. We can choose all other models we want to load with parameters, and it makes sense that loading TT is also set this way.

polinankoleva commented:
I completely disagree. The code is hard to read and understand with the variable. Moreover, a bunch of if blocks for a variable that is always true does not make any sense for me. Concerning the idea what will happen if we add a new model or if we change the requirements, we cannot exactly predict that. There is always a possibility some pieces of the code to be rewritten, added, removed and so on. Anyway, as I already mentioned, if both of you (@gschievelbein and @arrascue) really want to not remove the variable, I do not plan wasting more time into this discussion, just return/close the pull request.
